src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -184,12 +212,27 @@ bool MulticopterLandDetector::_get_ground_contact_state()
 	// an accurate in-air indication.
 	bool verticalMovement = fabsf(_vehicleLocalPosition.vz) > _params.maxClimbRate * armThresholdFactor;
 
+	// if we have a valid velocity setpoint and the vehicle is demanded to go down but no vertical movement present,
+	// we then can assume that the vehicle hit ground
+	bool in_descend = _is_climb_rate_enabled()
+			  && (_vehicleLocalPositionSetpoint.vz >= 0.9f * math::max(_params.landSpeed, 0.1f));
```

Comment:
Please add any fixes that make false positives for ground contact less likely when the land speed parameter is set too low.
For the context: We have seen twitches because of this but it's still safe according to my tests.